### PR TITLE
add build docs

### PR DESCRIPTION
**Pull request by pjfanning**
* copied from https://github.com/apache/incubator-pekko
* in anticipation that this repo will soon head to RC

### CODE REVIEW

**Review by mdedetrich — CHANGES_REQUESTED** [review]
Only Pekko needs dual JDK's, currently Pekko-Http only needs JDK 8.

**Comment by pjfanning** [comment]
> Only Pekko needs dual JDK's, currently Pekko-Http only needs JDK 8.

@mdedetrich I made the change to remove the mention of having 2 JDKs

File: README.md
Comment (mdedetrich) [inline README.md:34]: I woukd sayJDK 8 only (i.e. remove the later part) because Pekko Http build needs to produce JDK 8 bytecode. If you build with after JDK then JDK 8 users cannot run Pekko Http.


This can be resolved with the `-release` flag which pekko http uses but with conditionals which may not be relevant now (i.e. latest versions of Scala 2 fixes some problems) but the point is we need to be careful about this and verify it.

**Review by mdedetrich — CHANGES_REQUESTED** [review]
See comment

File: README.md
Comment (pjfanning) [reply README.md:34]: our pekko repo says JDK 11+

Comment (mdedetrich) [reply README.md:34]: The critical one, i.e. building jar artifacts is JDK 8 and we inherited that from Akka https://github.com/apache/incubator-pekko-http/blob/main/.github/workflows/publish.yml#L19 .

Also a distinction should be made between building release artifacts to validate release process and just working on the code locally.

I am on my phone atm so it's a bit hard to read entire thing in context but for me the important part is that it's clear that when creating release artifacts (for verification or doing the release yourself as a release manager) that you use JDK 8.

And it might well be that it cresting release artifacts for JDK 11 or higher is fine but after going through the process of validating/improving/clarifying this in root pekko i would rather be safe

Comment (pjfanning) [reply README.md:34]: this appears in our build.sbt

```
    (if (isJdk8) Seq.empty
     else if (scalaBinaryVersion.value == "2.12") Seq("-target:jvm-1.8")
     else Seq("-release", "8")),
```

Comment (mdedetrich) [reply README.md:34]: Yes I know but this was for older versions of Scala. `-target` in the latest versions of Scala is either deprecated or removed and ontop of the the behaviour of `-release` has changed.

Comment (pjfanning) [reply README.md:34]: I built pekko-http on my laptop with JDK11 and was then able to use the jars that I published to MavenLocal with JDK 8 builds of downstream projects. The classes appear to be Java 8 format.

Comment (mdedetrich) [reply README.md:34]: I am more concerned about stuff like unsafe or missing references to documentation. I'll check this properly when I get home

**Comment by mdedetrich** [comment]
@pjfanning So I haven't had proper time to check this, but the issue I am concerned about specifically is https://github.com/classgraph/classgraph/issues/403#issuecomment-586770739 and https://github.com/apache/incubator-pekko/pull/258/files#diff-c11a10bd9e773399e4272cd1ee5dff72545afb2f8bed23f2d1bbc49d4e1fc03fL60-L67 . JDK 9+ introduced a very subtle but possible breaking change from the JDK 8 standard library. If the problem is there, you won't hit at compile time, you hit them at runtime and only if you hit specific codepaths i.e. code that calls `byteBuffer.clear()` (and I briefly checked pekko-http and it does call this code).

To confirm that there isn't a problem I have to manually inspect the bytecode, just like I did here https://github.com/apache/incubator-pekko/pull/258#issuecomment-1477908162

**Comment by mdedetrich** [comment]
@pjfanning Okay so I have successfully run [jardiff](https://github.com/lightbend-labs/jardiff) on two versions of pekko-http, one built with JDK 8 and the other built with JDK 11 using

```
jardiff -g ~/github/diff-temp/ ~/github/incubator-pekko-http/http-core/target/scala-2.12/ ~/github/incubator-pekko-http-2/http-core/target/scala-2.12/
```

and as I suspected we have the problem I noted before

image
(this is with `HttpHeaderParser$.class`)

**Comment by mdedetrich** [comment]
@pjfanning I managed to fix the issue that I described earlier that will let you produce jars correctly on JDK 11, will create the PR soon (having dinner now)

**Comment by mdedetrich** [comment]
@pjfanning I have just created https://github.com/apache/incubator-pekko-http/pull/187 which allows building correct artifacts for pekko-http with JDK 11 so you can use the original documentation that you provided in this PR.

Once that PR is merged I can approve this one.

**Review by mdedetrich — APPROVED** [review]
Since https://github.com/apache/incubator-pekko-http/pull/187 is merged I will approve this PR, you may want to update it since I have also verified its safe to build with JDK 11 (newer versions than that I haven't checked yet).

File: README.md
Comment (jrudolph) [reply README.md:34]: The safe option is to build with JDK 8. The compatibility mode of JDK 11 will probably work most of the times but we ran into issues in the past, where the JDK signature storage that allows JDK 11 to figure out which methods are present in JDK 8 was not completely accurate.

**Pull request merged**
